### PR TITLE
Add test data preparing for list implementation

### PR DESCRIPTION
**Pull request by odsod**
Implementing support for lists will require some minor refactoring. Taking
things in steps...

- chore: tweak codecov GitHub Action
- chore: add list examples
- chore: extract protobq.Load implementation to load.go
- chore: add unit tests for list encoding
- chore: add integration test for lists

### CODE REVIEW

File: internal/examples/proto/src/einride/bigquery/public/v1/dogecoin_transaction.proto [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+option go_package = "go.einride.tech/internal/examples/proto/gen/einride/bigquery/public/v1;publicv1";
+
+import "google/protobuf/timestamp.proto";
+import "google/type/date.proto";
+
+// Protobuf schema for the BigQuery public table:
+//
+//  bigquery-public-data.crypto_dogecoin.transactions
+message DogecoinTransaction {
+  string hash = 1; // STRING REQUIRED
```
Comment (alfredgunnar) [inline internal/examples/proto/src/einride/bigquery/public/v1/dogecoin_transaction.proto:14]: If this is supposed to be required in the BQ schema, how is that captured in the proto definition?

File: encoding/protobq/integration_test.go [hunk 2]
```diff
@@ -95,6 +95,17 @@ func Test_Integration_PublicDataSets(t *testing.T) {
 			Limit:     10,
 			Message:   &publicv1.WhosOnFirstGeoJson{},
 		},
+
```
Comment (alfredgunnar) [inline encoding/protobq/integration_test.go:98]: Commit message says this doesn't pass. Is that correct?

File: internal/examples/proto/src/einride/bigquery/public/v1/dogecoin_transaction.proto (hunk 1 above)
Comment (odsod) [reply internal/examples/proto/src/einride/bigquery/public/v1/dogecoin_transaction.proto:14]: It's not - everything is optional in the protos.

File: encoding/protobq/integration_test.go (hunk 2 above)
Comment (odsod) [reply encoding/protobq/integration_test.go:98]: Yes, but it will in the next upcoming PR. :smile:

**Review by alfredgunnar — APPROVED** [review]

**Pull request merged**